### PR TITLE
Basic test suite added

### PR DESCRIPTION
**Pull request by crankycoder**
This adds basic S3 mocking for the coinstallation JSON file.  The keypath was incorrectly specified before and it has now been updated to match the ETL job.

The srgutil dependency has been bumped up to 0.1.5 to use python3  interface objects as keys into the Context to prevent improper string based object lookups.

A basic requirements.txt file has also been added, but without hashes.

### CODE REVIEW

File: taar_lite/recommenders/guid_based_recommender.py [hunk 1]
```diff
@@ -47,9 +52,13 @@ def can_recommend(self, client_data):
 
     def recommend(self, client_data, limit):
```
Comment (mlopatka) [inline taar_lite/recommenders/guid_based_recommender.py]: Can we please specify a default value of 4 if limit is not passed. Also, comment that justification as being driven by the AMO UX team's choice of how these will be surfaced. Currently 4 recs covers all of TAAR-lite's use cases.

File: tests/test_guid_based_recommender.py [hunk 2]
```diff
@@ -24,14 +28,47 @@ def test_ctx():
...
-    ctx['clock'] = Clock()
-    ctx['s3data'] = s3data
-    ctx['cache'] = JSONCache(ctx)
-    return ctx
+    return default_context()
 
 
+@mock_s3
 def test_recommender(test_ctx):
+    SAMPLE_DATA = {
```
Comment (mlopatka) [inline tests/test_guid_based_recommender.py]: Can we please replace the test guids with fake guids in the form "guid-01", "guid-02" so we are not leaking information from a real client query.

File: taar_lite/recommenders/guid_based_recommender.py (hunk 1 above)
Comment (crankycoder) [reply taar_lite/recommenders/guid_based_recommender.py]: Added a default of 4 for this.

File: tests/test_guid_based_recommender.py (hunk 2 above)
Comment (crankycoder) [reply tests/test_guid_based_recommender.py]: Fixed in new commit.

**Pull request merged**